### PR TITLE
Increasing automation of S2->S5

### PR DESCRIPTION
**Pull request by taylorbell57**
Hi Megan, I added a bit more of the functionality from S4 which allows multiple apertures from S3 to be analyzed with S4 and S5 to allow a parameter study without significant manual labour. I also simplified some aspects of the code (like automating the toffset value). I haven't run the code yet to see what breaks, but things should at least be pretty close to functional. I'll get back to work on this on the 29th

### CODE REVIEW

File: eureka/S5_lightcurve_fitting/s5_fit.py [hunk 1]
```diff
@@ -57,58 +57,120 @@ def fitJWST(eventlabel, s4_meta=None):
...
-        if s5_meta.isplots_S5 > 1:
-            lc_model.plot(s5_meta, draw=True)
+                # Set the intial fitting parameters
+                params = p.Parameters(param_file=meta.fit_par)
+                if meta.run_verbose:
+                    print(params)
+
+                # Subtract off the zeroth time value to avoid floating point precision problems when fitting for t0
+                t_offset = np.floor(meta.bjdtdb[0])
+                t_bjdtdb = meta.bjdtdb - t_offset
```
Comment (kevin218) [inline eureka/S5_lightcurve_fitting/s5_fit.py:140]: This should really be called `t_mjdtdb' since it is now a modified JD.

**Review by kevin218 — APPROVED** [review]

**Review by kevin218 — APPROVED** [review]

**Review by kevin218 — APPROVED** [review]

File: eureka/S5_lightcurve_fitting/fitters.py [hunk 2]
```diff
@@ -1,741 +1,512 @@
...
-        The model to fit
-    uncertainty: np.ndarray (optional)
-        The uncertainty on the (same shape) data
-    method: str
-        The name of the method to use
-    name: str
-        A name for the best fit model
-    verbose: bool
-        Print some stuff
+    lc: eureka.S5_lightcurve_fitting.lightcurve.LightCurve
```
Comment (meganmansfield) [inline eureka/S5_lightcurve_fitting/fitters.py:350]: I believe the lmfitter wasn't implemented yet, and after a bit of playing around I can't get the parameters in the right format. Perhaps someone who has used lmfit before can fix this later? For now, I just removed it as a S5 fitting option in the template ecf file.

**Review by meganmansfield — COMMENTED** [review]
I'm not sure exactly how the commenting on PRs process works, but I left a couple comments here. I'm also pushing some changes to the files, pretty much all just catching small typos. With those changes, I'm able to run lsq, emcee, and dynesty fits on S4 output without any problems.

File: eureka/S5_lightcurve_fitting/plots_s5.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    """
+    if type(fitter)!=str:
+        raise ValueError('Expected type str for fitter, instead received a {}'.format(type(fitter)))
+    
+    model_lc = model.eval()
+    residuals = (lc.flux - model_lc) #/ lc.unc
+    if meta.isplots_S5 >= 1:
+        model.plot(time=lc.time, draw=True)
+        
+        plt.figure(int('51{}_'.format(str(lc.channel).zfill(len(str(lc.nchannel))))), figsize=(8, 6))
```
[the remark this answers is not in the document]
Comment (taylorbell57) [reply eureka/S5_lightcurve_fitting/plots_s5.py]: I believe this is updated with the current code - let me know if it isn't though

File: eureka/S5_lightcurve_fitting/fitters.py (hunk 2 above)
Comment (taylorbell57) [reply eureka/S5_lightcurve_fitting/fitters.py:358]: Fair enough, I'm just about to test it and may come to the same conclusion

Comment (taylorbell57) [reply eureka/S5_lightcurve_fitting/fitters.py:350]: Yup, lmfitter is currently still very broken - let's do as you say and hide it from users for now and we can work on resolving this later

File: eureka/S5_lightcurve_fitting/plots_s5.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (taylorbell57) [reply eureka/S5_lightcurve_fitting/plots_s5.py]: As far as I can tell, this is indeed fixed with the most recent code

File: eureka/S5_lightcurve_fitting/fitters.py (hunk 2 above)
Comment (kevin218) [reply eureka/S5_lightcurve_fitting/fitters.py:343]: Not sure if I've ever gotten lmfit to work properly. 🤷‍♂️

**Pull request merged**